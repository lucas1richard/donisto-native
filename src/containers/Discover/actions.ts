/*
 *
 * Discover actions
 *
 */

import DiscoverConstants from './constants';

export type TDiscoverActions =
  IDefaultAction
;

interface IDefaultAction {
  type: DiscoverConstants.DEFAULT_ACTION
}

export const defaultAction = (): IDefaultAction => ({
  type: DiscoverConstants.DEFAULT_ACTION
});
