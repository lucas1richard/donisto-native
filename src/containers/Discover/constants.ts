/*
 *
 * Discover constants
 *
 */

enum DiscoverConstants {
  DEFAULT_ACTION = 'app/Discover/DEFAULT_ACTION'
}

export default DiscoverConstants;
