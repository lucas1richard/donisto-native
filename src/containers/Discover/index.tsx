import * as React from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';
import SwipeCards from 'react-native-swipe-cards';
import logger from 'utilities/logger';
import { Entypo } from '@expo/vector-icons';
import theme from 'theme/iftheme';
import Txt from 'components/Txt';

interface ICardProps {
  image: string;
  name: string;
}

class Card extends React.Component<ICardProps> {
  render() {
    return (
      <View style={styles.card}>
        <Image style={styles.thumbnail} source={{ uri: this.props.image }} />
        <Text style={styles.text}>This is card {this.props.name}</Text>
      </View>
    );
  }
}

class NoMoreCards extends React.Component {
  render() {
    return (
      <View style={styles.noMoreCards}>
        <Text>No more cards</Text>
      </View>
    );
  }
}

const cards = [
  { name: '1', image: 'https://source.unsplash.com/random/200x200?0' },
  { name: '2', image: 'https://source.unsplash.com/random/200x200?1' },
  { name: '3', image: 'https://source.unsplash.com/random/200x200?2' },
  { name: '4', image: 'https://source.unsplash.com/random/200x200?3' },
  { name: '5', image: 'https://source.unsplash.com/random/200x200?4' },
  { name: '6', image: 'https://source.unsplash.com/random/200x200?5' },
  { name: '7', image: 'https://source.unsplash.com/random/200x200?6' },
  { name: '8', image: 'https://source.unsplash.com/random/200x200?7' },
  { name: '9', image: 'https://source.unsplash.com/random/200x200?8' }
];

const cards2 = [
  { name: '10', image: 'https://source.unsplash.com/random/200x200?9' },
  { name: '11', image: 'https://source.unsplash.com/random/200x200?10' },
  { name: '12', image: 'https://source.unsplash.com/random/200x200?11' },
  { name: '13', image: 'https://source.unsplash.com/random/200x200?12' }
];

export interface IDiscoverProps {}
interface IDiscoverState {
  cards: any[];
  outOfCards: boolean
}

export default class Discover extends React.Component<IDiscoverProps, IDiscoverState> {
  constructor(props: IDiscoverProps) {
    super(props);
    this.state = {
      cards: cards,
      outOfCards: false
    };
  }

  handleYup = (/* card: any */) => {
    logger.log('yup', 'Discover/index.tsx');
  }

  handleNope = (/* card: any */) => {
    logger.log('nope', 'Discover/index.tsx');
  }

  cardRemoved = (index: number) => {
    let CARD_REFRESH_LIMIT = 3;

    if (this.state.cards.length - index <= CARD_REFRESH_LIMIT + 1) {
      logger.log(`There are only ${this.state.cards.length - index - 1} cards left.`, 'Discover/index.tsx');

      if (!this.state.outOfCards) {
        logger.log(`Adding ${cards2.length} more cards`, 'Discover/index.tsx');

        this.setState({
          cards: this.state.cards.concat(cards2),
          outOfCards: true
        });
      }
    }
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <View style={{
          height: 60,
          borderBottomWidth: 1,
          borderBottomColor: '#d3d3d3',
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'space-between',
          paddingHorizontal: 20
        }}>
          <Entypo
            name="plus"
            color={theme.primary[500]}
            size={25}
          />
          <Txt style={{ fontSize: 22, fontWeight: '600' }}>
            Discover
          </Txt>
          <Entypo
            name="magnifying-glass"
            color={theme.primary[500]}
            size={25}
          />
        </View>
        <SwipeCards
          cards={this.state.cards}
          loop={false}
          renderCard={(cardData: ICardProps) => <Card {...cardData} />}
          renderNoMoreCards={() => <NoMoreCards />}
          showYup={true}
          showNope={true}
          handleYup={this.handleYup}
          handleNope={this.handleNope}
          cardRemoved={this.cardRemoved}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  card: {
    alignItems: 'center',
    borderRadius: 5,
    overflow: 'hidden',
    borderColor: 'grey',
    backgroundColor: 'white',
    borderWidth: 1,
    elevation: 1
  },
  thumbnail: {
    width: 300,
    height: 300
  },
  text: {
    fontSize: 20,
    paddingTop: 10,
    paddingBottom: 10
  },
  noMoreCards: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
