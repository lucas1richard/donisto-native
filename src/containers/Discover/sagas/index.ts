import { all } from 'redux-saga/effects';

export function* discoverSaga() {
  yield all([
  ]);
}

export default discoverSaga;
