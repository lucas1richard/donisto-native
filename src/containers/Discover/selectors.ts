import { createSelector } from 'reselect';

export const selectDiscoverDomain = () => (state) => state.discover;

export const makeSelectDiscover = () => createSelector(
  selectDiscoverDomain(),
  (substate) => substate
);
