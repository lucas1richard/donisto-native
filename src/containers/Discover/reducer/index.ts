import DiscoverConstants from '../constants';
import { TDiscoverActions } from '../actions';

const initialState = {};

function discoverReducer(state = initialState, action: TDiscoverActions) {
  switch (action.type) {
    case DiscoverConstants.DEFAULT_ACTION:
      return state;
    default:
      return state;
  }
}

export default discoverReducer;
