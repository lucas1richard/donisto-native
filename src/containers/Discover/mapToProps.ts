import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { createStructuredSelector } from 'reselect';
import { makeSelectDiscover } from './selectors';

const mapStateToProps = createStructuredSelector({
  Discover: makeSelectDiscover()
});

function mapDispatchToProps(dispatch: Dispatch) {
  return {
    dispatch
  };
}

export default connect(mapStateToProps, mapDispatchToProps);
