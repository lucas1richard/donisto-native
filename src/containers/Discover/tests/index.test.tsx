import * as React from 'react';
import { shallow } from 'enzyme';

import Discover,  { IDiscoverProps } from '../index';

describe('<Discover />', () => {
  const renderComponent = (props: IDiscoverProps) => shallow(<Discover {...props} />);
  it('Expect to have unit tests specified', () => {
    expect(renderComponent({})).toMatchSnapshot();
  });
});
