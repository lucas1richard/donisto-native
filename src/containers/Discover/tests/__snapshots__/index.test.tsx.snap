// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Discover /> Expect to have unit tests specified 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Discover />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#d3d3d3",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
              "paddingHorizontal": 20,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="rgb(251, 100, 65)"
            name="plus"
            size={25}
          />
          <Txt
            color="default"
            style={
              Object {
                "fontSize": 22,
                "fontWeight": "600",
              }
            }
          >
            Discover
          </Txt>
          <Icon
            allowFontScaling={false}
            color="rgb(251, 100, 65)"
            name="magnifying-glass"
            size={25}
          />
        </Component>,
        <SwipeCards
          allowGestureTermination={true}
          cardKey="key"
          cardRemoved={[Function]}
          cards={
            Array [
              Object {
                "image": "https://source.unsplash.com/random/200x200?0",
                "name": "1",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?1",
                "name": "2",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?2",
                "name": "3",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?3",
                "name": "4",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?4",
                "name": "5",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?5",
                "name": "6",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?6",
                "name": "7",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?7",
                "name": "8",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?8",
                "name": "9",
              },
            ]
          }
          dragY={true}
          handleMaybe={[Function]}
          handleNope={[Function]}
          handleYup={[Function]}
          hasMaybeAction={false}
          loop={false}
          maybeText="Maybe!"
          nopeText="Nope!"
          onClickHandler={[Function]}
          onDragRelease={[Function]}
          onDragStart={[Function]}
          onLoop={[Function]}
          renderCard={[Function]}
          renderNoMoreCards={[Function]}
          showMaybe={true}
          showNope={true}
          showYup={true}
          smoothTransition={false}
          stack={false}
          stackDepth={5}
          stackOffsetX={25}
          stackOffsetY={0}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            }
          }
          yupText="Yup!"
        />,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Icon
              allowFontScaling={false}
              color="rgb(251, 100, 65)"
              name="plus"
              size={25}
            />,
            <Txt
              color="default"
              style={
                Object {
                  "fontSize": 22,
                  "fontWeight": "600",
                }
              }
            >
              Discover
            </Txt>,
            <Icon
              allowFontScaling={false}
              color="rgb(251, 100, 65)"
              name="magnifying-glass"
              size={25}
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "borderBottomColor": "#d3d3d3",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingHorizontal": 20,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "color": "rgb(251, 100, 65)",
              "name": "plus",
              "size": 25,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Discover",
              "color": "default",
              "style": Object {
                "fontSize": 22,
                "fontWeight": "600",
              },
            },
            "ref": null,
            "rendered": "Discover",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "color": "rgb(251, 100, 65)",
              "name": "magnifying-glass",
              "size": 25,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowGestureTermination": true,
          "cardKey": "key",
          "cardRemoved": [Function],
          "cards": Array [
            Object {
              "image": "https://source.unsplash.com/random/200x200?0",
              "name": "1",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?1",
              "name": "2",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?2",
              "name": "3",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?3",
              "name": "4",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?4",
              "name": "5",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?5",
              "name": "6",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?6",
              "name": "7",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?7",
              "name": "8",
            },
            Object {
              "image": "https://source.unsplash.com/random/200x200?8",
              "name": "9",
            },
          ],
          "dragY": true,
          "handleMaybe": [Function],
          "handleNope": [Function],
          "handleYup": [Function],
          "hasMaybeAction": false,
          "loop": false,
          "maybeText": "Maybe!",
          "nopeText": "Nope!",
          "onClickHandler": [Function],
          "onDragRelease": [Function],
          "onDragStart": [Function],
          "onLoop": [Function],
          "renderCard": [Function],
          "renderNoMoreCards": [Function],
          "showMaybe": true,
          "showNope": true,
          "showYup": true,
          "smoothTransition": false,
          "stack": false,
          "stackDepth": 5,
          "stackOffsetX": 25,
          "stackOffsetY": 0,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "justifyContent": "center",
          },
          "yupText": "Yup!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "paddingHorizontal": 20,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="rgb(251, 100, 65)"
              name="plus"
              size={25}
            />
            <Txt
              color="default"
              style={
                Object {
                  "fontSize": 22,
                  "fontWeight": "600",
                }
              }
            >
              Discover
            </Txt>
            <Icon
              allowFontScaling={false}
              color="rgb(251, 100, 65)"
              name="magnifying-glass"
              size={25}
            />
          </Component>,
          <SwipeCards
            allowGestureTermination={true}
            cardKey="key"
            cardRemoved={[Function]}
            cards={
              Array [
                Object {
                  "image": "https://source.unsplash.com/random/200x200?0",
                  "name": "1",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?1",
                  "name": "2",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?2",
                  "name": "3",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?3",
                  "name": "4",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?4",
                  "name": "5",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?5",
                  "name": "6",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?6",
                  "name": "7",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?7",
                  "name": "8",
                },
                Object {
                  "image": "https://source.unsplash.com/random/200x200?8",
                  "name": "9",
                },
              ]
            }
            dragY={true}
            handleMaybe={[Function]}
            handleNope={[Function]}
            handleYup={[Function]}
            hasMaybeAction={false}
            loop={false}
            maybeText="Maybe!"
            nopeText="Nope!"
            onClickHandler={[Function]}
            onDragRelease={[Function]}
            onDragStart={[Function]}
            onLoop={[Function]}
            renderCard={[Function]}
            renderNoMoreCards={[Function]}
            showMaybe={true}
            showNope={true}
            showYup={true}
            smoothTransition={false}
            stack={false}
            stackDepth={5}
            stackOffsetX={25}
            stackOffsetY={0}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
            yupText="Yup!"
          />,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Icon
                allowFontScaling={false}
                color="rgb(251, 100, 65)"
                name="plus"
                size={25}
              />,
              <Txt
                color="default"
                style={
                  Object {
                    "fontSize": 22,
                    "fontWeight": "600",
                  }
                }
              >
                Discover
              </Txt>,
              <Icon
                allowFontScaling={false}
                color="rgb(251, 100, 65)"
                name="magnifying-glass"
                size={25}
              />,
            ],
            "style": Object {
              "alignItems": "center",
              "borderBottomColor": "#d3d3d3",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
              "paddingHorizontal": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": false,
                "color": "rgb(251, 100, 65)",
                "name": "plus",
                "size": 25,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Discover",
                "color": "default",
                "style": Object {
                  "fontSize": 22,
                  "fontWeight": "600",
                },
              },
              "ref": null,
              "rendered": "Discover",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowFontScaling": false,
                "color": "rgb(251, 100, 65)",
                "name": "magnifying-glass",
                "size": 25,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowGestureTermination": true,
            "cardKey": "key",
            "cardRemoved": [Function],
            "cards": Array [
              Object {
                "image": "https://source.unsplash.com/random/200x200?0",
                "name": "1",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?1",
                "name": "2",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?2",
                "name": "3",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?3",
                "name": "4",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?4",
                "name": "5",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?5",
                "name": "6",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?6",
                "name": "7",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?7",
                "name": "8",
              },
              Object {
                "image": "https://source.unsplash.com/random/200x200?8",
                "name": "9",
              },
            ],
            "dragY": true,
            "handleMaybe": [Function],
            "handleNope": [Function],
            "handleYup": [Function],
            "hasMaybeAction": false,
            "loop": false,
            "maybeText": "Maybe!",
            "nopeText": "Nope!",
            "onClickHandler": [Function],
            "onDragRelease": [Function],
            "onDragStart": [Function],
            "onLoop": [Function],
            "renderCard": [Function],
            "renderNoMoreCards": [Function],
            "showMaybe": true,
            "showNope": true,
            "showYup": true,
            "smoothTransition": false,
            "stack": false,
            "stackDepth": 5,
            "stackOffsetX": 25,
            "stackOffsetY": 0,
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
            },
            "yupText": "Yup!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
